server.py

```diff
@@ -13,183 +12,193 @@
-
-	def send_command(self, client, command):
-		command = pickle.dumps(command)
-		msg = bytes(f"{len(command):<{HEADER_SIZE}}", BYTE_ENCODING) + command  # add fixed length header to message
-		# print(msg[HEADER_SIZE:])
-		client.send(msg)  # command to client
-
-	def buffer_message(self, client):
-		msg_flag = True
-		buffer = b''
-		while True:
-			pkt = client.recv(100)  # buffer the received data
-			if not pkt:
-				client.close
-				break
-			if msg_flag:
-				# print("new message length:", pkt[:HEADER_SIZE])
-				msg_length = int(pkt[:HEADER_SIZE])
-				msg_flag = False
-
-			buffer += pkt
-			full_msg_length = len(buffer)
-			# logging.info("message buffer length:" + str(full_msg_length))
-
-			if len(buffer) - HEADER_SIZE == msg_length:
-				logging.info("Full message received with length:" + str(full_msg_length))
-
-				# print(buffer[HEADER_SIZE:])
-				server_message = pickle.loads(buffer[HEADER_SIZE:])
-				msg_flag = True
-				buffer = b''
-				# logging.info("-------------------------------------------------")
-				# logging.info("Received message: %s", server_message)
-
-				return server_message  # parsed_message
-
-
-class GameServer(Messenger):
-
-	# default constructor
-	def __init__(self, srv_ip, srv_port):
-		self.host_ip = srv_ip
-		self.port = srv_port
-
-		self.player_id = 1
-		self.whose_turn = 1  # pointer to current player taking turn
-		self.game_over = False
-
-		self.new_player_id = 1
-		self.players = []  # keep track of players
+class QueryStatus(Enum):
+    SERVER_TO_CLIENT = 1  # Information is being sent from Executive Logic to Server to Client
+    CLIENT_TO_SERVER = 2  # Information is being sent from Client to Server to Executive Logic
+    STANDBY = 3  # No information is being sent right now
 
 
-		# SETUP SUBSYSTEMS
-		self.board = Board()
-		self.executive_logic = ExecutiveLogic()
-		self.score_keeper = ScoreKeeper()
-		self.wheel = Wheel()
-		self.ui = UserInterface()
+class MessageType(Enum):
+    """
+	Enum of all message types that may be sent between executive <-> server <-> client.
+	Part of the Message class. Always accompanied by a list of arguments, which is different based on the message type.
 
-		self.num_spins = 0  # TODO: this should really be held by the ExecutiveLogic
+	Each message type will be sent by the Executive (requesting information), with some arguments.
+	A message of the same type will then be sent by the Client (responding), with different arguments.
+	"""
+    EMPTY = 0  # No message.
+    # Request  Args: 	[]
+    # Response Args:	[]
 
-	def host_game(self):
-		server = socket.socket(socket.AF_INET, socket.SOCK_STREAM)  # setup server socket
-		server.bind((self.host_ip, self.port))
-		logging.info("server starting a game, listening for players on port: " + str(self.port))
-		server.listen(NUM_PLAYERS - 1)  # listen for 2 connections. 3 player game
+    PLAYER_ID = 1  # Called by server the first time it connects to a client. Assigns that client its unique player ID.
```

Comment:
I noticed `player_id` is 1+ in the gameserver, but 0, 1 or 2 in the executive logic. So the first player to join gets a message the their `player_id` is 1. We should make sure these match. I think it's good to start with player 0.